docker/Dockerfile

```diff
@@ -1,44 +1,39 @@
-FROM python:3.7.2-alpine3.8
-
-ENV PYTHONUNBUFFERED=1
-ENV PYTHONDONTWRITEBYTECODE=1
+FROM python:3.7.3-slim as base
 ENV PATH="/root/.poetry/bin:$PATH"
 ENV FLASK_APP=search
-
 WORKDIR /srv/app
-
-RUN rm -rf /var/cache/apk/* && \
-    rm -rf /tmp/* && \
-    apk update && \
-    apk add --no-cache \
-    bash \
-    grep \
-    libxslt-dev
-
-RUN apk add --no-cache --virtual .build-dependencies \
-    curl \
-    && \
-    curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python && \
-    poetry config settings.virtualenvs.create false && \
-    apk del .build-dependencies
-
 COPY ./pyproject.toml ./poetry.lock ./
-
-ARG env
-ENV FLASK_ENV=${env}
-
-ARG dev
-ARG simulate_removal
-RUN apk add --no-cache --virtual .build-dependencies \
-    gcc \
-    git \
-    linux-headers \
-    musl-dev \
-    && \
-    poetry install ${dev} && \
-    apk del ${simulate_removal} .build-dependencies
-
 COPY search/ search/
```

Comment:
It is not the topic of this PR, but this will need to be separated into the poetry files (`pyproject.toml`, `poetry.lock`?) and the rest of the source code; the goal being that changing a single `.py` file doesn't result in reinstalling poetry when rebuilding locally.